### DISCUSSION TITLE
Metal/FCPX support on airyxOS?

### DISCUSSION THREAD

**Discussion by wizoomer95** [ravynsoft/ravynos#176] [Ideas]
Hello,

I just came across this project from the recent XDA-developers article.  I applaud you for your effort.  I previously proposed a similar open-source operating system concept [elsewhere](https://www.insanelymac.com/forum/topic/344972-courtlandos-an-open-source-operating-system-idea-developers-wanted/), albeit it planned to use Darwin as opposed to FreeBSD.  As I state in that forum post, I'm not a developer, so between that and the lack of replies, I just never got started on it.  But it does bring up another question for my potential use case.

Don't get me wrong: Despite my pessimism in that forum post linked above, I love Apple Silicon as much as the next person.  I've used my M1 Mac Mini for a while now, and I've been eyeing an M1 Max MacBook Pro for a while now.  I also am familiar with Hackintoshing and have built several over the years.  Currently, while I enjoy the synergy across the Apple ecosystem, the one thing that's keeping me locked to using macOS as my daily driver is the fact my NLE of choice is Final Cut Pro X.  I'd love to see FCPX run on an open-source operating system like this, if only to prove I don't need to stay on macOS just to continue to use it.

I understand that FCPX (and I'm sure countless other graphics-intensive macOS applications) use the Metal API for graphics acceleration.  While I know and understand macOS application support for airyxOS is not there yet, and that any possible implementation of Metal would likely be light years down the pipeline, I figured I would ask if it would be possible to implement Metal support on airyxOS in the future?  I really hope this project succeeds regardless.  I'd love to learn how to code even if I only use it to contribute to this!  I even plugged this project in a follow-up post to the aforementioned forum thread.  I wish you all the best in your endeavors!

(Side note: How _exactly_ do you pronounce airyxOS?)

**Comment by mszoek** [comment]
Hi and thanks for the post! Metal is probably inevitable at some point, but you're right that it is not really on my radar in any serious way yet. We can keep this open or turn it into an Issue for future tracking :)

I pronounce it "AIR-icks Oh Ess". It was meant to combine airyness and unix.

**Comment by andrewcrook** [comment]
Could be a compatability layer as to get support on AMD and Nvidia.
Would like OpenGL and Vulkan support i.e get the things macOS should have or keep.

**Reply by mszoek** [reply]
100% it would be a compatibility shim exporting the Metal APIs on top of something like Vulkan or GL. Or at least that's my initial thinking on how to tackle it.
